pkg/util/defaults/defaults.go

```diff
@@ -44,5 +44,10 @@ const (
 	installDefaultKamelets = true
 )
 
+// Maps base image variants to their respective digests
+var baseImageVariantMapping = map[string]string{
```

Comment:
We cannot hardcode configuration values. Beside, this file should not be edited as it is regenerated automatically at every build execution.

Reply:
What you see here is the result of Makefile code generation. Perhaps this file should be in .gitignore

In Makefile we get the digest like this ... 

```
ARM64_DIGEST := $(shell docker manifest inspect eclipse-temurin:17 | jq -r '.manifests[] | select(.platform.architecture == "arm64") | .digest')
```

True, this approach fixes the base image digest at build time instead of dynamically loading the correct variant at maven build time - I could not find the source for that. Perhaps we can still get this into the code base for now and have a separate issues for (OLM, Helm, Kustomize)

Alternatively, we could do the above lookup (every time) when the base image is actually needed, which I assume would then work for all install methods

Reply:
This may hijack a little bit the issue but this comment about runtime lookup of the right base image make me thinking if we should do it for every architecture and having the `IntegrationPlatform` reflecting the images in its status.

The reason for that is something we never have discussed but it should be possible to have a mixed kubernetes set-up where some nodes are amd64 and some arm64 so you may have some kits working for some nodes and not for some others.